### ISSUE TITLE
Default values for Array and Dictionary in optional arguments in GDScript function show up as `<unknown>` in syntax help

### ISSUE THREAD

**Issue by Yagich** [godotengine/godot#78191]
### Godot version

4.1.beta1

### System information

Linux/Manjaro

### Issue description

When writing code in the default code editor, the syntax help popup shows the argument names and types, as well as default values, if any. However, if the help is for a script function, the default values for Arrays and Dictionaries will be shown as `<unknown>`. This does not happen for built-in functions or arguments of primitive types:

![image](https://github.com/godotengine/godot/assets/24468819/364eb7cf-0053-44fe-a4b3-b87a8512f55f)
![image](https://github.com/godotengine/godot/assets/24468819/cd2391c4-9c18-46f6-8295-1c54facfe8d7)
![image](https://github.com/godotengine/godot/assets/24468819/0ae81f23-cf3f-4e69-965b-271fbae24e4e)
![image](https://github.com/godotengine/godot/assets/24468819/ee039abe-9610-4231-b3db-7ea0c40c30bb)
![image](https://github.com/godotengine/godot/assets/24468819/8554225b-0801-476a-9850-72188d3ba43e)

### Steps to reproduce

1. Make a function that takes an optional Array or Dictionary as an argument
2. Write the function and trigger the syntax help (<kbd>Ctrl</kbd>+<kbd>Space</kbd> inside parentheses)
3. Observe the error as shown on above screenshots

### Minimal reproduction project

[unknown-default-param-value.zip](https://github.com/godotengine/godot/files/11736437/unknown-default-param-value.zip)

**Comment by dalexeev** [comment]
In theory, we can add a value to the tooltip and documentation, but only if it is a constant expression[^1], because the current documentation system does not allow to add arbitrary strings representing source code expressions.

Also we need to limit the length of the array/dictionary string representation.

Please add the `topic:gdscript` label.

[^1]: This is not entirely true, since arrays and dictionaries are passed by reference (const arrays and dictionaries are implemented slightly differently, they are marked read-only with the `make_read_only()` method).

**Comment by bitwise-aiden** [comment]
I stumbled across this issue and wondered if there is room for a quality of life update in lieu of constexpr. After peaking through the code a bit I came up with the following solution. 

https://github.com/godotengine/godot/assets/60748675/bbfe0a45-9baa-46fe-bdbc-1e9b0093cd75

The filled case should probably be `..` instead of `...` to keep inline with the syntax of the match statement.

**Comment by vnen** [comment]
@bitwise-aiden do you plan making a pull request with your solution?

**Comment by bitwise-aiden** [comment]
@vnen Yup, just put up a PR for it now. It's not a comprehensive solution, but I think that it still provides some quality of life to developers.
